### ISSUE TITLE
What is the best way to watch events?

### ISSUE THREAD

**Issue by kn1g** [MetaMask/metamask-extension#3311]
I always get in trouble with things like `var filter = web3.eth.filter({fromBlock: '5139365', toBlock: 'latest'}); filter.watch(function(_err, _resp){ ... `

What is the recommended way to watch for new blocks when using the MetaMask provider?

**Comment by costa595** [comment]
I've faced with the same problem, watching events is a headache

**Comment by psyantyst** [comment]
TL;DR: use testrpc instead.

I can also confirm that watching events using MetaMask doesn't work. Events are fired correctly, but are not intercepted by MetaMask, even when using correct filters, fromBlock, toBlock, etc. 

I spent a whole day figuring out where the problem lies. Then I disabled MetaMask, tried with testrpc instead and everything worked perfectly.

**Comment by costa595** [comment]
@psyantyst thank you for your reply, but for example: once I load my contract to the main network will events work fine MetaMask?

**Comment by psyantyst** [comment]
@costa595 just as I posted my reply, I realized that my solution doesn't apply if you are interested in listening on a non-dev network. But since OP did not specify the type of network, my reply is still a valid solution/workaround for those who are still developing.

I just tested MetaMask on my privately deployed Ethereum network (a few nodes using geth) and it seems that MetaMask does not intercept blockchain events. So we still need a solution for production environments.
